SU2_CFD/include/variables/CVariable.hpp

```diff
@@ -937,6 +937,22 @@ class CVariable {
    */
   inline virtual su2double GetMassFraction(unsigned long iPoint, unsigned long val_Species) const { return 0.0; }
 
+  /*!
+   * \brief A virtual member.
+   * \param[in] iPoint - Point index.
+   * \param[in] val_Species - Index of species s.
+   * \return Value of the mass fraction of species s.
+   */
+  inline virtual su2double GetEve(unsigned long iPoint) const { return 0.0; }
+
+  /*!
+   * \brief A virtual member.
+   * \param[in] iPoint - Point index.
+   * \param[in] val_Species - Index of species s.
+   * \return Value of the mass fraction of species s.
+   */
+  inline virtual su2double* GetEnthalpys(unsigned long iPoint) { return nullptr; }
```

Comment:
Move to CNEMOEulerVariable please, since it looks this only needs to be visible in the solver

Reply:
Sorry, should I move the virtual function def from CVariable -> CNEMOEulerVariable, or should I move the function def from CNEMONSVariable -> CNEMOEulerVariable?

Reply:
What error? In FVMFlowSolverBase the nodes are of type CNEMOEulerVariable, so this should work.

Reply:
This is the error I get on compiling:
`In file included from ../SU2_CFD/src/solvers/CNSSolver.cpp:32:
../SU2_CFD/src/solvers/../../include/solvers/CFVMFlowSolverBase.inl:2592:53: error: no member named ‘GetEnthalpys’ in ‘CEulerVariable’
    const auto hs           = nodes->GetEnthalpys(iPoint);
                       ~~~~~ ^
../SU2_CFD/src/solvers/CNSSolver.cpp:36:16: note: in instantiation of member function ‘CFVMFlowSolverBase<CEulerVariable, ENUM_REGIME::COMPRESSIBLE>::Friction_Forces’ requested here
template class CFVMFlowSolverBase<CEulerVariable, ENUM_REGIME::COMPRESSIBLE>;`

Reply:
Dohh :facepalm: sorry my bad. You can leave it in CVariable then.